wordpress-mobile/AztecEditor-iOS
# New Format Bar

I think this is ready for a first review. All the functionality isn't present just yet, but here's where we're at:

* Bar displays a list of items, with a set of extra 'overflow' items that can be toggled using the '...' ellipsis button.
* If the screen width is too small to fit all of the default items, then some of them will be hidden and can be revealed with the ellipsis button. For example, on an iPhone SE we miss out the link button on the end:

<img width="359" alt="screen shot 2017-06-12 at 16 40 11" src="https://user-images.githubusercontent.com/4780/27042113-ef50c174-4f8d-11e7-9f47-aec5d71ed5e9.png">

But it gets shown on wider screens:

<img width="589" alt="screen shot 2017-06-12 at 16 40 23" src="https://user-images.githubusercontent.com/4780/27042123-f5952548-4f8d-11e7-930f-10327c55eef1.png">

* The bar has limited support for separators between items – we only use it for the first item in the list, so it's not particularly fleshed out at this stage. Works for our needs though. Both the top divider and dividers between items are colored with the `dividerTintColor` property.

* I've rearranged the items to match our designs.
* When the bar is 'expanded', it becomes scrollable.

**Not implemented in this branch:**

* We don't update the heading button's image to reflect different heading levels yet.
* We don't show popovers for heading selection yet.
* Selecting different types of lists isn't supported.
* We currently don't remember the last 'expanded / collapsed' state when viewing the editor.

**To test:**

* Ensure all the expected items show up in the bar, and it matches the designs.
* Test on a smaller device that the last item gets removed from the bar, but shows up if you rotate or expand.
* Check the expand / collapse interaction works as expected.
* Check all buttons on the bar function.

Needs review: @jleandroperez – would you mind? Sorry it's a bit of a big one, but also better we review now before it gets any bigger!
